src/app/shared/code-editor-list/code-editor-list.component.ts

```diff
@@ -76,73 +61,20 @@ export class CodeEditorListComponent {
   }
 
   /**
-   * Get the file type enum
-   * @return {string} The file type enum
-   */
-  getFileType(filepath) {
-    if (this.fileType === 'descriptor') {
-      if (this.descriptorType) {
-        if (this.descriptorType === ToolDescriptor.TypeEnum.NFL) {
-          if (filepath === this.NEXTFLOW_CONFIG_PATH) {
-            return 'NEXTFLOW_CONFIG';
-          } else {
-            return 'NEXTFLOW';
-          }
-        } else {
-          return 'DOCKSTORE_' + this.descriptorType;
-        }
-      } else {
-        return 'DOCKSTORE_CWL';
-      }
-    } else if (this.fileType === 'testParam') {
-      if (this.descriptorType) {
-        if (this.descriptorType === ToolDescriptor.TypeEnum.NFL) {
-          return 'NEXTFLOW_TEST_PARAMS';
-        } else {
-          return this.descriptorType + '_TEST_JSON';
-        }
-      } else {
-        return 'CWL_TEST_JSON';
-      }
-    } else if (this.fileType === 'dockerfile') {
-      return 'DOCKERFILE';
-    } else {
-      return null;
-    }
-  }
-
-  /**
-   * Get the default path extension
-   * @return {string}the default path extension
-   */
-  getDefaultPath() {
-    if (this.fileType === 'descriptor') {
-      if (this.descriptorType) {
-        if (this.descriptorType === this.DescriptorType.NFL) {
-          return '.nf';
-        } else {
-          return '.' + this.descriptorType.toLowerCase();
-        }
-      } else {
-        return '.cwl';
-      }
-    } else if (this.fileType === 'testParam') {
-      return '.json';
-    } else if (this.fileType === 'dockerfile') {
-      return 'Dockerfile';
-    }
-  }
-
-  /**
+   * TODO: Fix the execution of this function.  This function is being executed a bajillion times.
    * Returns true if path is the primary descriptor, false otherwise
-   * @param  path Path to check
-   * @return {boolean}      Is path for primary descriptor
+   *
+   * @static
+   * @param {string} path Path to check
+   * @returns {boolean} Is path for primary descriptor
+   * @memberof CodeEditorListComponent
    */
   isPrimaryDescriptor(path: string): boolean {
-    return path === '/Dockstore.cwl' || path === '/Dockstore.wdl' || path === this.NEXTFLOW_CONFIG_PATH || path === this.NEXTFLOW_PATH;
+    return CodeEditorListService.isPrimaryDescriptor(path);
```

Comment:
You said in your PR comment that a lot of this logic is odd and you're keeping it that way for now, so it is probably beyond the scope of this, but without grokking the whole flow, what happens if the primary descriptor is 'alignment.wdl'?

Reply:
Everything is messed up, webservice too